tools/markdown-to-html/docs-marked-renderer.ts

```diff
@@ -52,12 +52,20 @@ export class DocsMarkdownRenderer extends Renderer {
    *  `<!-- example(name) -->` turns into `<div material-docs-example="name"></div>`
    */
   html(html: string) {
-    html = html.replace(exampleCommentRegex, (_match: string, name: string) =>
-      `<div material-docs-example="${name}"></div>`
-    );
+        html = html.replace(exampleCommentRegex, (_match: string, content: string) =>
+        {
+            const parsedContent = JSON.parse(content);
+            const exampleName = parsedContent['example']
+            const file = parsedContent['file'];
+            const lines = parsedContent['lines'];
+            const expanded = !!parsedContent['expanded'];
```

Comment:
Might be worth considering destructuring here. e.g. `const {exampleName, file, lines, expanded} = <..>` (and the boolean coercion later inline)